src/main/java/org/micromanager/acqj/internal/Engine.java

```diff
@@ -512,6 +533,21 @@ private void acquireImages(final AcquisitionEvent event,
             correspondingEvent.acquisition_.addToImageMetadata(ti.tags);
 
             correspondingEvent.acquisition_.addToOutput(ti);
+
+            }
+         }
+         if (future != null) {
+            try {
+               future.get(1000, TimeUnit.MILLISECONDS);
+            } catch (InterruptedException e) {
+               e.printStackTrace();
+               core_.logMessage("Interrupted while waiting for the future");
+            } catch (ExecutionException e) {
+               e.printStackTrace();
+               core_.logMessage("Exception during future execution");
+            } catch (TimeoutException e) {
+               e.printStackTrace();
+               core_.logMessage("Timeout while waiting for the future");

```

Comment:
What situation should this timeout in? I would think this shoudl hang indefinitely until executed or throws an exception